frontend/src/components/Experiment/Experiment.js

```diff
@@ -65,12 +65,26 @@ const Experiment = ({ match }) => {
         setActions(newActions);
     }, [loadState, setActions]);
 
-    // Start first_round when experiment and partipant have been loaded
     useEffect(() => {
-        if (session.current) {
-            setSessionStore(session.current);
+        if (!experiment || !participant || !playlist) {
+            return;
+        }
+        try { 
+            createSession({
+                experiment,
+                participant,
+                playlist,
+            }).then(data => {
+                setSession(data.session);
+                setSessionStore(data.session);
```

Comment:
Why do we need these both still?

Reply:
This is for "historical reasons" - I thought to leave adapting the child components of `Experiment`, some of which still expect a `session` prop, for later.